src/pages/authPages/Login.tsx

```diff
@@ -1,8 +1,42 @@
 import React, { FC } from 'react';
 import { Box, TextField, Button, Typography, Container, Paper } from '@mui/material';
 import NavigationBar from '../homePage/NavigationBar';
+import * as yup from "yup"
+import { useAuthContext } from '../../context/AuthContext';
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom';
+
+interface LoginData {
+    email?: string,
```

Comment:
czemu opcjonalny? czemu dublujesz interface z contextu?
Moze warto zrobic folder "types" i tam trzymac pliki z typami?

Reply:
taki plik będzie :) tez o nim myślałem teraz chciałem zrobić żeby na lekcji działało.
Case z opcjonalnym mailem był spowodowany tym ze właśnie zjadłem .required() i resolver mi krzyczał o ten mail dlatego było to rozwiązanie doraźne 😅